main.c

```diff
@@ -177,11 +212,15 @@ static void pointer_handle_button(void *data, struct wl_pointer *wl_pointer,
 		uint32_t serial, uint32_t time, uint32_t button,
 		uint32_t button_state) {
 	struct slurp_seat *seat = data;
+	if (seat->touch_selection.has_selection) {
+	  return;
+	}
 
 	seat->button_state = button_state;
 
 	switch (button_state) {
 	case WL_POINTER_BUTTON_STATE_PRESSED:
+		seat->pointer_selection.has_selection = true;
```

Comment:
This should be taken care of by `handle_selection_start`. `handle_selection_start` should take a `struct slurp_selection *` as an argument.